apps/prairielearn/src/server.ts

```diff
@@ -1964,6 +1962,12 @@ export async function startServer() {
 
   server.timeout = config.serverTimeout;
   server.keepAliveTimeout = config.serverKeepAliveTimeout;
+
+  if ((import.meta as any).env?.DEV) {
+    return server;
+  }
+
+  // In production, startup the server normally
```

Comment:
In general, I'd consider things like this to be a code smell - development should behave as closely to production as possible. Though having worked on HMR systems before, I know that sometimes things need special casing. So it's not necessarily wrong/bad; I just want to make sure we understand why changes like this are necessary.

In this case, I definitely don't. Why _shouldn't_ we start the server in dev mode? We still need a server to run.

Reply:
In this case, vite is responsible for starting up the server, in middleware mode. More details in this comment: https://github.com/PrairieLearn/PrairieLearn/pull/11914#issuecomment-2851706325